src/coordinator/tsdb/remote/compressed_codecs.go

```diff
@@ -0,0 +1,250 @@
+	segment, err := br.Segment()
+	if err != nil {
+		return nil, err
+	}
+	return &rpc.Segment{
+		Head: segment.Head.Bytes(),
+		Tail: segment.Tail.Bytes(),
+	}, nil
+}
+
+// RPCFromSeriesIterator converts a seriesIterator to a grpc series
+func RPCFromSeriesIterator(it encoding.SeriesIterator) (*rpc.Series, error) {
+	replicas := it.Replicas()
+	compressedReplicas := make([]*rpc.CompressedValuesReplica, 0, len(replicas))
+	for _, replica := range replicas {
+		replicaSegments := make([]*rpc.Segments, 0, len(replicas))
+		readers := replica.Readers()
+		for next := true; next; next = readers.Next() {
+			segments := &rpc.Segments{}
+			l, _, _ := readers.CurrentReaders()
+			// NB(arnikola) If there's only a single reader, the segment has been merged
+			// otherwise, multiple unmerged segments exist.
+			if l == 1 {
+				br := readers.CurrentReaderAt(0)
+				segment, err := blockReaderToSegment(br)
+				if err != nil {
+					return nil, err
+				}
+				segments.Merged = segment
+			} else {
+				unmerged := make([]*rpc.Segment, 0, l)
+				for i := 0; i < l; i++ {
+					br := readers.CurrentReaderAt(i)
+					segment, err := blockReaderToSegment(br)
+					if err != nil {
+						return nil, err
+					}
+					unmerged = append(unmerged, segment)
+				}
+				segments.Unmerged = unmerged
+			}
+			replicaSegments = append(replicaSegments, segments)
+		}
+		compressedReplicas = append(compressedReplicas, &rpc.CompressedValuesReplica{
+			Segments: replicaSegments,
+		})
+	}
+
+	start := xtime.ToNanoseconds(it.Start())
+	end := xtime.ToNanoseconds(it.End())
+
+	tagIter := it.Tags()
+	tags := make([]*rpc.CompressedTag, 0, tagIter.Remaining())
+	for tagIter.Next() {
+		tag := tagIter.Current()
+		tags = append(tags, &rpc.CompressedTag{
+			Name:  tag.Name.Bytes(),
+			Value: tag.Value.Bytes(),
+		})
+	}
+	err := tagIter.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	compressedDatapoints := &rpc.CompressedDatapoints{
+		Namespace: it.Namespace().Bytes(),
+		StartTime: start,
+		EndTime:   end,
+		Replicas:  compressedReplicas,
+		Tags:      tags,
+	}
+
+	return &rpc.Series{
+		Id:         it.ID().Bytes(),
+		Compressed: compressedDatapoints,
+	}, nil
+}
+
+func segmentToBlockReader(
+	seg *rpc.Segment,
+	opts checked.BytesOptions,
+	iteratorPools encoding.IteratorPools,
+) xio.BlockReader {
+	var checkedBytesWrapperPool xpool.CheckedBytesWrapperPool
+	if iteratorPools != nil {
+		checkedBytesWrapperPool = iteratorPools.CheckedBytesWrapper()
+	}
+
+	var head, tail checked.Bytes
+	if checkedBytesWrapperPool != nil {
+		head = checkedBytesWrapperPool.Get(seg.Head)
+		tail = checkedBytesWrapperPool.Get(seg.Tail)
+	} else {
+		head = checked.NewBytes(seg.Head, opts)
+		tail = checked.NewBytes(seg.Tail, opts)
+	}
+
+	segment := ts.NewSegment(head, tail, ts.FinalizeNone)
+	segmentReader := xio.NewSegmentReader(segment)
+
+	return xio.BlockReader{
+		SegmentReader: segmentReader,
+		Start:         time.Time{},
+		BlockSize:     0,
+	}
+}
+
+var (
```

Comment:
when does this get initialized ? Maybe put this inside an init method ? Don't want this to run in cases when we are not using rpc